ingestion/src/metadata/ingestion/source/pipeline/dbtcloud/metadata.py

```diff
@@ -0,0 +1,322 @@
+                        continue
+
+                    for dest in model.parentsSources or []:
+                        from_entity = self.metadata.get_by_name(
+                            entity=Table,
+                            fqn=fqn.build(
+                                metadata=self.metadata,
+                                entity_type=Table,
+                                table_name=dest.name,
+                                database_name=dest.database,
+                                schema_name=dest.dbtschema,
+                                service_name=dbservicename,
+                            ),
+                        )
+
+                        if from_entity is None:
+                            continue
+
+                        yield Either(
+                            right=AddLineageRequest(
+                                edge=EntitiesEdge(
+                                    fromEntity=EntityReference(
+                                        id=from_entity.id,
+                                        type="table",
+                                    ),
+                                    toEntity=EntityReference(
+                                        id=to_entity.id,
+                                        type="table",
+                                    ),
+                                    lineageDetails=lineage_details,
+                                )
+                            )
+                        )
+
+        except Exception as exc:
+            yield Either(
+                left=StackTraceError(
+                    name=pipeline_details.name,
+                    error=f"Wild error ingesting pipeline lineage {pipeline_details} - {exc}",
+                    stackTrace=traceback.format_exc(),
+                )
+            )
+
+    def get_pipelines_list(self) -> Iterable[DBTJob]:
+        """
+        Get List of all pipelines
+        """
+        try:
+            yield from self.client.get_jobs()
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(f"Failed to get pipeline list due to : {exc}")
+        return None
+
+    def get_pipeline_name(self, pipeline_details: DBTJob) -> str:
+        """
+        Get Pipeline Name
+        """
+        try:
+            return pipeline_details.name
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(f"Failed to get pipeline name due to : {exc}")
+
+        return None
+
+    def yield_pipeline_status(
+        self, pipeline_details: DBTJob
+    ) -> Iterable[Either[OMetaPipelineStatus]]:
+        """
+        Get Pipeline Status
+        """
+        try:
+            task_status = [
+                TaskStatus(
+                    name=str(task.id),
+                    executionStatus=STATUS_MAP.get(task.state, StatusType.Pending),
+                    startTime=Timestamp(
+                        datetime_to_ts(
```

Comment:
Do we need to convert the timestamp here in milliseconds?

Reply:
`datetime_to_ts` converts a given date to a timestamp as an Int in milliseconds

Reply:
use [datetime.timestamp](https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp) and multiply by 1000.